Lib/asyncio/unix_events.py

```diff
@@ -1259,19 +1266,33 @@ def remove_child_handler(self, pid):
             return False
 
     def attach_loop(self, loop):
+        """
+        This registers the SIGCHLD signal handler.
+
+        This method can only be called from the main thread.
+        """
         # Don't save the loop but initialize itself if called first time
         # The reason to do it here is that attach_loop() is called from
         # unix policy only for the main thread.
         # Main thread is required for subscription on SIGCHLD signal
+        if loop is None or self._saved_sighandler is not None:
+            return
+
+        self._loop = loop
```

Comment:
Part of why I was saying to remove this line is to honor the code comment 7 lines above:

```python
# Don't save the loop but initialize itself if called first time
```